src/waveform/renderers/waveformmark.cpp

```diff
@@ -127,14 +136,242 @@ bool WaveformMark::contains(QPoint point, Qt::Orientation orientation) const {
     // Without some padding, the user would only have a single pixel width that
     // would count as hovering over the WaveformMark.
     float lineHoverPadding = 5.0;
-    int position;
-    if (orientation == Qt::Horizontal) {
-        position = point.x();
-    } else {
-        position = point.y();
+    if (orientation == Qt::Vertical) {
+        point = QPoint(point.y(), m_breadth - point.x());
     }
-    bool lineHovered = m_linePosition >= position - lineHoverPadding &&
-            m_linePosition <= position + lineHoverPadding;
+    bool lineHovered = m_linePosition >= point.x() - lineHoverPadding &&
+            m_linePosition <= point.x() + lineHoverPadding;
 
     return m_label.area().contains(point) || lineHovered;
 }
+
+// Helper class to calculate the geometry and fontsize needed by generateImage
```

Comment:
technically this is a struct not a class, so let's call it that

Reply:
Done